homeassistant/components/comfoconnect/fan.py

```diff
@@ -99,27 +101,30 @@ def supported_features(self) -> int:
     @property
     def percentage(self) -> int | None:
         """Return the current speed percentage."""
-        speed = self._ccb.data.get(SENSOR_FAN_SPEED_MODE)
-        if speed is None:
+        if self.current_speed is None:
             return None
-        return ranged_value_to_percentage(SPEED_RANGE, speed)
+        return ranged_value_to_percentage(SPEED_RANGE, self.current_speed)
 
     @property
     def speed_count(self) -> int:
         """Return the number of speeds the fan supports."""
         return int_states_in_range(SPEED_RANGE)
 
     def turn_on(
-        self, speed: str = None, percentage=None, preset_mode=None, **kwargs
+        self,
+        speed: str | None = None,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs,
     ) -> None:
         """Turn on the fan."""
         self.set_percentage(percentage)
 
-    def turn_off(self, **kwargs) -> None:
+    def turn_off(self, **kwargs: Any) -> None:
         """Turn off the fan (to away)."""
         self.set_percentage(0)
 
-    def set_percentage(self, percentage: int):
+    def set_percentage(self, percentage: int | None) -> None:
```

Comment:
It's better to convert `None` to a specific percentage integer in `turn_on` instead of accepting `None` here.

Reply:
Here's the signature of the parent method:
https://github.com/home-assistant/core/blob/0656407561dd93a5d90749f9a1b8379a1c2ddacb/homeassistant/components/fan/__init__.py#L278

Reply:
Huh. You are right. Will change this.

Reply:
I've changed this to use a percentage of `1` in `turn_on` if there is no percentage passed. This will map to a low speed then as it was before.